packages/zosjobs/src/api/DownloadJobs.ts

```diff
@@ -68,12 +68,23 @@ export class DownloadJobs {
         this.log.debug("Downloading all spool content for job %s(%s)", parms.jobname, parms.jobid);
 
         const jobFiles = await GetJobs.getSpoolFiles(session, parms.jobname, parms.jobid);
+        // Maintain a mapping of step names and associated DD names to detect duplicate step names
+        const usedStepNames: {[key: string]: string[]} = {};
         for (const file of jobFiles) {
+            let uniqueDDName = file.ddname;
+            if (usedStepNames[file.stepname] != null) {
+                let index = 1;
+                while (usedStepNames[file.stepname].indexOf(uniqueDDName) !== -1) {
+                    uniqueDDName = `${file.ddname}(${index})`;
```

Comment:
Not sure what the best naming convention is to make filenames unique. I followed the Windows convention of appending `(1)`, `(2)`, etc., but perhaps a different convention like appending `_2`, `_3`, etc. would be better.

Reply:
Perhaps pass it an as a part and set your default?  API users could adjust it when programmatically using it.  Someday if someone cared you could surface it as an option on the CLI?

Reply:
Sounds good but I'm not clear how it would be passed in. Perhaps the suffix could be specified with a wildcard, like `(*)` or `_*`, where `*` would be replaced by a number?

Reply:
oh yeah good point haha - maybe you should leave it as you have it and leave that stuff for another day

Reply:
I would suggest against implementing with parenthesis just for the sake of problems that arise on *nix systems. Granted, they can be easily mitigated if you are certain you want to implement in this way.

```
[andrew@test-VM GitHub]$ touch "file.txt"
[andrew@test-VM GitHub]$ mv file.txt file(1).txt
bash: syntax error near unexpected token `('
[andrew@test-VM GitHub]$ mv file.txt "file(1).txt"
[andrew@test-VM GitHub]$ 
```

Reply:
Nevermind, I just came to the realization that modern web browsers and file explorers on linux do take the approach of appending `(copy)` or `(1)` to duplicate files.